cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/distributed/AbstractProgramTwillRunnable.java

```diff
@@ -356,14 +360,38 @@ public void error(Throwable cause) {
     }
   }
 
+  /**
+   * On an authorization-enabled cluster, if caching is enabled too, updates the cache in the
+   * {@link AuthorizationEnforcementService} with the privileges of the user running the program.
+   */
+  private void updateAuthCache() {
```

Comment:
This is only called once, before the program starts up. It is here, to pass the ``Principal`` running the program from ``AbstractProgramTwillRunnable`` to ``AuthorizationEnforcementService``.